django/db/models/sql/query.py

```diff
@@ -1627,6 +1627,9 @@ def clear_limits(self):
         """Clear any existing limits."""
         self.low_mark, self.high_mark = 0, None
 
+    def has_limit_one(self):
+        return (self.high_mark - self.low_mark) == 1
```

Comment:
Is this going to crash if `self.high_mark is None`?

Reply:
Yes, I've fixed that.